src/intl/time-durations.js

```diff
@@ -0,0 +1,235 @@
+// @flow
+
+import moment from 'moment'
+import 'moment/locale/cs'
+import 'moment/locale/de'
+import 'moment/locale/es'
+import 'moment/locale/fr'
+import 'moment/locale/it'
+import 'moment/locale/ja'
+import 'moment/locale/ko'
+import 'moment/locale/pt-br'
+import 'moment/locale/zh-cn'
+
+import 'moment-duration-format'
+
+export function setupMomentTranslations (locale: string, defaultLocale: string) {
+  if (locale === 'aa') {
+    moment.defineLocale('aa', {})
+  }
+
+  const chosen = moment.locale([ locale, defaultLocale ])
+  console.log(`Locale being used by moment: ${chosen}`)
+
+  //
+  // Extend the moment locale object for moment-duration-format:
+  //   https://github.com/jsmreese/moment-duration-format#extending-moments-locale-object
+  //
+  const translations = require('./translated-time-durations.json')
+  if (translations[locale]) {
+    const t:{ [messageId: string]: string } = translations[locale]
+
+    moment.updateLocale(locale, {
+      durationLabelsStandard: {
+        S: t.durationLabelStandard_S || timeDurations.durationLabelStandard_S.message,
+        SS: t.durationLabelStandard_SS || timeDurations.durationLabelStandard_SS.message,
+        s: t.durationLabelStandard__s || timeDurations.durationLabelStandard_s.message,
```

Comment:
Good catch, thanks!



> Typo in key name (double underscore).

Good catch, thanks!

> Manual mapping is quite error prone - it would be nice to add more validation here (either runtime or via type system).

Yeah, but stuff does need to be mapped, and validations aren't automatic either, and types also aren't automatic....I don't really know of an easy way to do this kind of thing in javascript.  We can chat about ideas for these kinds of improvements in future.